mac

```diff
@@ -188,6 +140,75 @@ zsh_config_homebrew() {
   append_to_zshrc "eval \"\$($HOMEBREW_PREFIX/bin/brew shellenv)\""
 }
 
+install_asdf() {
+  fancy_echo "Configuring asdf version manager ..."
+  if [ ! -d "$HOME/.asdf" ]; then
+    brew install asdf
+    append_to_zshrc "source $(brew --prefix asdf)/asdf.sh" 1
+  fi
+}
+
+add_or_update_asdf_plugin() {
+  local name="$1"
+  local url="$2"
+
+  if ! asdf plugin-list | grep -Fq "$name"; then
+    asdf plugin-add "$name" "$url"
+  else
+    asdf plugin-update "$name"
+  fi
+}
+
+install_asdf_language() {
+  local language="$1"
+  local version
+  version="$(asdf list-all "$language" | grep -v "[a-z]" | tail -1)"
+
+  if ! asdf list "$language" | grep -Fq "$version"; then
+    asdf install "$language" "$version"
+    asdf global "$language" "$version"
+  fi
+}
+
+install_asdf_ruby() {
+  # shellcheck disable=SC1090
+  . "$(brew --prefix asdf)/asdf.sh"
+  add_or_update_asdf_plugin "ruby" "https://github.com/asdf-vm/asdf-ruby.git"
+
+  fancy_echo "Installing latest Ruby ..."
+  install_asdf_language "ruby"
+  gem update --system
+  number_of_cores=$(sysctl -n hw.ncpu)
+  bundle config --global jobs $((number_of_cores - 1))
+}
+
+install_asdf_nodejs() {
+  # shellcheck disable=SC1090
+  . "$(brew --prefix asdf)/asdf.sh"
+  add_or_update_asdf_plugin "nodejs" "https://github.com/asdf-vm/asdf-nodejs.git"
+
+  fancy_echo "Installing latest Node ..."
+  install_asdf_language "nodejs"
+}
+
+install_asdf_erlang() {
+  # shellcheck disable=SC1090
+  . "$(brew --prefix asdf)/asdf.sh"
+  add_or_update_asdf_plugin "erlang" "https://github.com/asdf-vm/asdf-erlang.git"
+
+  fancy_echo "Installing latest Erlang ..."
+  install_asdf_language "erlang"
+}
+
+install_asdf_elixir() {
+  # shellcheck disable=SC1090
+  . "$(brew --prefix asdf)/asdf.sh"
+  add_or_update_asdf_plugin "elixir" "https://github.com/asdf-vm/asdf-elixir.git"
+
+  fancy_echo "Installing latest Elixir ..."
+  install_asdf_language "elixir"
```

Comment:
For elixir, I think we need to pass the version or update the sorting order in the  `version="$(asdf list-all "$language" | grep -v "[a-z]" | tail -1)"`, because when I ran that command, it returns `1.13.4`, with erlang, it returns 25.0. As a recommendation, they recommend using the Elixir `1.13.4-otp-25` install of just only `1.13.4`

![image](https://user-images.githubusercontent.com/11751745/170912716-c3042adb-6a0e-4470-ab45-b0be6124cb86.png)


![image](https://user-images.githubusercontent.com/11751745/170912839-a43c1c3b-e0b2-4378-9e16-4a92d23ba0ff.png)

Reply:
Indeed, on my setup, I have `1.13.4` instead of `1.13.4-otp-25` 🙈 So great catch. 

Made the change in [d12fba1](https://github.com/nimblehq/laptop/pull/34/commits/d12fba1abb6f4b4cbae72bbeda9eab916a823ff5)

Here is the tested version:

![Screen Shot 2565-05-31 at 11 37 57](https://user-images.githubusercontent.com/696529/171094104-592f4795-6add-47ff-a8c1-1f839bdd7145.png)